python/pyspark/sql/tests/test_connect_compatibility.py

```diff
@@ -23,155 +23,165 @@
+
+        # Compare the actual missing properties with the expected ones
+        self.assertEqual(
+            classic_only_properties,
+            expected_missing_properties,
+            f"{cls_name}: Unexpected missing properties in Connect: {classic_only_properties}",
         )
 
-        # Expected missing properties for Column (if any, replace with actual values)
-        expected_missing_properties_for_column = set()
-
-        # Column properties comparison
-        compare_property_lists(
-            ClassicColumn, ConnectColumn, "Column", expected_missing_properties_for_column
-        )
+    def _check_missing_methods(self, classic_cls, connect_cls, cls_name, expected_missing_methods):
+        """Check for expected missing methods between classic and connect classes."""
+        classic_methods = self._get_public_methods(classic_cls)
+        connect_methods = self._get_public_methods(connect_cls)
 
-        # Expected missing properties for SparkSession
-        expected_missing_properties_for_spark_session = {"sparkContext", "version"}
+        # Identify missing methods
+        classic_only_methods = set(classic_methods.keys()) - set(connect_methods.keys())
 
-        # SparkSession properties comparison
-        compare_property_lists(
-            ClassicSparkSession,
-            ConnectSparkSession,
-            "SparkSession",
-            expected_missing_properties_for_spark_session,
+        # Compare the actual missing methods with the expected ones
+        self.assertEqual(
+            classic_only_methods,
+            expected_missing_methods,
+            f"{cls_name}: Unexpected missing methods in Connect: {classic_only_methods}",
         )
 
-    def test_missing_methods(self):
-        def check_missing_methods(classic_cls, connect_cls, cls_name, expected_missing_methods):
-            """Check for expected missing methods between classic and connect classes."""
-            classic_methods = self.get_public_methods(classic_cls)
-            connect_methods = self.get_public_methods(connect_cls)
-
-            # Identify missing methods
-            classic_only_methods = set(classic_methods.keys()) - set(connect_methods.keys())
-
-            # Compare the actual missing methods with the expected ones
-            self.assertEqual(
-                classic_only_methods,
-                expected_missing_methods,
-                f"{cls_name}: Unexpected missing methods in Connect: {classic_only_methods}",
-            )
-
-        # Expected missing methods for DataFrame
-        expected_missing_methods_for_dataframe = {
-            "inputFiles",
-            "isLocal",
-            "semanticHash",
-            "isEmpty",
-        }
-
-        # DataFrame missing method check
-        check_missing_methods(
-            ClassicDataFrame, ConnectDataFrame, "DataFrame", expected_missing_methods_for_dataframe
+    def check_compatibility(
```

Comment:
FYI: Now this function is a single point of all compatibility check.

Before this, we should use `compare_method_signatures`, `compare_property_lists` and `check_missing_methods` respectively so there was a bunch of duplicated test codes.